src/config/index.ts

```diff
@@ -0,0 +1,130 @@
+import jsPlugin from '@eslint/js';
+import type { ESLint, Linter } from 'eslint';
+import prettierConfig from 'eslint-config-prettier';
+import importPlugin from 'eslint-plugin-import';
+import reactPlugin from 'eslint-plugin-react';
+import reactHooksPlugin from 'eslint-plugin-react-hooks';
+import globals from 'globals';
+import tsEslint from 'typescript-eslint';
+import rules from '../rules';
+
+export type ConfigOptions = Readonly<
+  Partial<{
+    /**
+     * TypeScript の設定ファイル (languageOptions.parserOptions.project)
+     * デフォルト: ./tsconfig.json
+     */
+    tsProject: boolean | string | string[] | undefined;
```

Comment:
Off topic: I think it is a good idea to make the default value of `tsProject` to `true`.

https://typescript-eslint.io/blog/parser-options-project-true/

Reply:
Yep, I know.

The reason I used `./tsconfig.json` here is that the current `.eslintrc` style config uses `./tsconfig.json` and I wanted to keep the configs as same as possible.

https://github.com/hatena/eslint-config-hatena/blob/547609d55009e084cca0b78ca7d54f87523724ae/src/classic/typescript.ts#L21